core/src/main/kotlin/org/evomaster/core/search/gene/regex/CharacterClassEscapeRxGene.kt

```diff
@@ -44,9 +43,14 @@ class CharacterClassEscapeRxGene(
     }
 
     override fun standardMutation(randomness: Randomness, apc: AdaptiveParameterControl, allGenes: List<Gene>) {
+        if (value=="") {
+            // if standardMutation was invoked before calling to randomize
+            // then do a randomize first
+            randomize(randomness, forceNewValue = true, allGenes = allGenes)
+        }
 
         value = when(type){
-            "d" -> ((value.toInt() + randomness.choose(listOf(1,-1) + 10)) % 10).toString()
+            "d" -> ((Math.abs(value.toInt() + randomness.choose(listOf(1,-1) + 10))) % 10).toString()
```

Comment:
why was `abs` necessary? can it ever happen that it is negative?

Reply:
yes. I found cases where the value is set to 0 and the randomness chose -1.
The result of -1 % 10 is -1.

Reply:
yes, but that is the reason for why there is a `+ 10`, so a `0` on which `-1` is applied become a `9` :)

Reply:
I wrote the code to:
            "d" -> ((value.toInt() + randomness.choose(listOf(1,-1)) + 10) % 10).toString()